src/brains2/src/sim/sim.cpp

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2024, Tudor Oancea, Matteo Berthet
+#include "brains2/sim/sim.hpp"
+#include <cmath>
+#include "acados_c/sim_interface.h"
+#include "brains2/common/math.hpp"
+#include "generated/acados_sim_solver_dyn6.h"
+#include "generated/acados_sim_solver_kin6.h"
+#include "generated/dyn6_accels.h"
+#include "generated/kin6_accels.h"
+
+using namespace brains2::sim;
+using namespace brains2::common;
+
+const std::array<std::string, 4> Sim::SimErrorStrings = {
+    "SAMPLING_TIME_UPDATE_ERROR",
+    "ACADOS_SOLVER_ERROR",
+    "ACCELS_FUNCTION_ERROR",
+    "NANS_IN_RESULT",
+};
```

Comment:
Put this in .hpp so that we keep the relationship between the error and it's associated string (with the enum class SimError )

Reply:
that's impossible because it's a static variable https://stackoverflow.com/a/5019896

Reply:
what about static inline or namespaces ? https://stackoverflow.com/questions/10876930/should-one-never-use-static-inline-function

Reply:
Your post seem related to static methods, not variables. But it could be a simpler solution

Reply:
https://stackoverflow.com/q/47819719

Reply:
?

Reply:
I think something like this could be simpler:
```cpp
static inline std::string to_string(SimError error) {
    switch (error) {
        case SimError::SAMPLING_TIME_UPDATE_ERROR:
            return "SAMPLING_TIME_UPDATE_ERROR";
        case SimError::ACADOS_SOLVER_ERROR:
            return "ACADOS_SOLVER_ERROR";
        case SimError::ACCELS_FUNCTION_ERROR:
            return "ACCELS_FUNCTION_ERROR";
        case SimError::NANS_IN_RESULT:
            return "NANS_IN_RESULT";
        default:
            return "UNKNOWN_ERROR";
    }
}
```

Reply:
bahahahahahaha my bad

Reply:
Well it's even better I think because is reduce again the possibility of mismatch but static inline for a variable works as well.